test/java/seedu/addressbook/commands/SortCommandTest.java

```diff
@@ -0,0 +1,61 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Test;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TestUtil;
+import seedu.addressbook.util.TypicalPersons;
+
+public class SortCommandTest {
+    private TypicalPersons td = new TypicalPersons();
+
+    private AddressBook typicalAddressBook = td.getTypicalAddressBook();
+    private List<ReadOnlyPerson> listWithTypicalPersons = Arrays.asList(td.bill, td.amy, td.candy, td.dan);
+    private List<ReadOnlyPerson> secondListWithTypicalPersons = Arrays.asList(td.dan, td.candy, td.amy, td.bill);
+    private List<ReadOnlyPerson> thirdListWithTypicalPersons = Arrays.asList(td.candy, td.dan, td.bill, td.amy);
+    
+    @Test
+    public void execute_validIndex_returnsPersonDetails() {
+        assertViewBehavior(new SortCommand(), typicalAddressBook, listWithTypicalPersons);
+        assertViewBehavior(new SortCommand(), typicalAddressBook, secondListWithTypicalPersons);
+        assertViewBehavior(new SortCommand(), typicalAddressBook, thirdListWithTypicalPersons);
```

Comment:
Consider adding test cases where the person's names differ only at the second character and beyond. Example: Aaron, Adam.